src/main/kotlin/com/gamingbacklog/api/gamingbacklogapi/clients/ExternalAPICall.kt

```diff
@@ -0,0 +1,29 @@
+package com.gamingbacklog.api.gamingbacklogapi.clients
+
+import org.springframework.context.annotation.Bean
+import org.springframework.stereotype.Component
+import java.net.URI
+import java.net.http.HttpClient
+import java.net.http.HttpRequest
+import java.net.http.HttpResponse
+
+@Component
+class ExternalAPICall {
+  fun postExternalCall(
+    uri: String,
+    body: String? = null,
+    headers: HashMap<String, String>? = null,
+  ): HttpResponse<String> {
+    val requestBody = if (body != null)  HttpRequest.BodyPublishers.ofString(body) else HttpRequest.BodyPublishers.noBody()
+    val requestBuilder: HttpRequest.Builder? = HttpRequest.newBuilder()
+      .uri(URI.create(uri))
+      .POST(requestBody)
+    if (headers != null) {
+      for ((key, value) in headers) {
+        requestBuilder?.header(key, value)
+      }
+    }
+    val request = requestBuilder?.build()
+    return HttpClient.newHttpClient().send(request, HttpResponse.BodyHandlers.ofString())
+  }
+}
```

Comment:
Do we have any linting/formatting set up for Kotlin to handle newlines and things like that?

Reply:
I started working on it, but it got really complicated really quickly as the kotlin linter I was using seemed very finnicky with newer versions of Java and Windows. We have https://gaming-backlog.atlassian.net/browse/GB-50 where we can finalize this and get it working.